src/doc/src/reference/semver.md

```diff
@@ -2152,7 +2159,33 @@ curl = { version = "0.4.31", optional = true }
 # ..curl removed
 ```
 
+```toml
+# MINOR CHANGE
+#
+# This example shows how to avoid breaking changes with optional dependencies.
+
+###########################################################
+# Before
+[dependencies]
+curl = { version = "0.4.31", optional = true }
+
+[features]
+networking = ["dep:curl"]
```

Comment:
I think its fine for us to suggest this be treated as minor.  The main risk is rust-lang/rust#109005 as people might be seeing the internal names and using them.  I actually saw someone [mislead by this](https://www.youtube.com/watch?v=jVoEA7qmN8c&t=8s) when demoing the compiler error.